src/aspire/source/__init__.py

```diff
@@ -363,12 +362,10 @@ def whiten(self, noise_filter):
         whiten_filter = PowerFilter(noise_filter, power=-0.5)
 
         logger.info('Transforming all CTF Filters into Multiplicative Filters')
-        unique_filters = set(self.filters)
-        for f in unique_filters:
-            f_new = copy(f)
-            f.__class__ = MultiplicativeFilter
-            f.__init__(f_new, whiten_filter)
-
+        for f_ind in range(len(self.unique_filters)):
```

Comment:
Again, if you're only interested in getting the filter, not its index, simply iterate through `self.unique_filters`. No need for an index.

Reply:
Done.